functions/src/sections/Section.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Seat, Section } from './Section.model';
+// TODO: replace once main repo is updated.
+import { UVicCourseScraper } from '@vikelabs/uvic-course-scraper/dist/index';
+import { db } from '../db/firestore';
+import { subjectCodeExtractor } from '../shared/subjectCodeExtractor';
+import { Term } from '../constants';
+import { InvalidSubjectCodeError } from '../errors/errors';
+
+export class SectionsService {
+  public async getSections(
+    term: string,
+    subject: string,
+    code: string
+  ): Promise<Section[]> {
+    const sections = await UVicCourseScraper.getCourseSections(
+      term,
+      subject,
+      code
+    );
+
+    return sections;
+  }
+
+  public async getSectionSeats(
+    term: Term,
+    subject: string,
+    code: string
+  ): Promise<Seat[]> {
+    // get term, subject, code to crn mappings from db.
+    const doc = await db.sectionMappings
+      .doc(SectionsService.constructSectionKey(term, subject, code))
+      .get();
+
+    const data = doc.data();
+
+    if (!data) {
+      throw new InvalidSubjectCodeError('Section Seats Not Found');
```

Comment:
First, do you mean invalid subject and code pair?

This error can be thrown when the subject-code (course) is actually valid but the course just isn't in the db that term because there are no offerings.

Reply:
We don't have a way to differentiate I guess by assuming a hit/miss in the database. We would have to make empty entries for valid queries but I don't know if that's a good idea or not...

Reply:
In terms of the `error`, it was me just testing and leaving something to work off of.